sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateViewAsSelect.scala

```diff
@@ -44,26 +47,36 @@ private[hive] case class CreateViewAsSelect(
   override def run(sqlContext: SQLContext): Seq[Row] = {
     val hiveContext = sqlContext.asInstanceOf[HiveContext]
 
-    if (hiveContext.catalog.tableExists(tableIdentifier)) {
-      if (allowExisting) {
+    hiveContext.catalog.tableExists(tableIdentifier) match {
+      case true if allowExisting =>
```

Comment:
While you are changing this part, how about we add a comment to show the sql command that will trigger this case? (also add the comment for `orReplace` branch?)